test/compiler.test.js

```diff
@@ -1043,27 +1044,47 @@ describe('compiler', function() {
           'errorHandler');
       });
 
-    // FIXME: [rfeng] The following test is disabled until
-    // https://github.com/strongloop/loopback-boot/issues/73 is fixed
-    it.skip('resolves modules relative to appRootDir', function() {
-        var HANDLER_FILE = 'node_modules/handler/index.js';
-        appdir.writeFileSync(
-          HANDLER_FILE,
-          'module.exports = function(req, res, next) { next(); }');
+    it('resolves modules relative to appRootDir', function() {
+      var HANDLER_FILE = 'node_modules/handler/index.js';
+      appdir.writeFileSync(
+        HANDLER_FILE,
+        'module.exports = function(req, res, next) { next(); }');
 
-        appdir.writeConfigFileSync('middleware.json', {
-          'initial': {
-            'handler': {}
-          }
-        });
+      appdir.writeConfigFileSync('middleware.json', {
+        'initial': {
+          'handler': {}
+        }
+      });
 
-        var instructions = boot.compile(appdir.PATH);
+      var instructions = boot.compile(appdir.PATH);
 
-        expect(instructions.middleware.middleware[0]).have.property(
-          'sourceFile',
-          appdir.resolve(HANDLER_FILE));
+      expect(instructions.middleware.middleware[0]).have.property(
+        'sourceFile',
+        appdir.resolve(HANDLER_FILE));
+    });
+
+    it('explores middleware function in appRootDir', function() {
```

Comment:
Please use a more descriptive name, e.g. "prefers appRootDir over node_modules for middleware".